### PR TITLE
Refactor battery enums

### PR DESCRIPTION
**Pull request by brtietz**
Updates the GUI given the changes in SSC PR https://github.com/NREL/ssc/pull/602, and expands the scope of the custom weather file forecast to AC batteries and PV smoothing. Fixes https://github.com/NREL/SAM/issues/676 BTM forecasting will be another PR.

Forecast will run in a callback when the weather file is specified and option 2 is selected. SSC will throw an error if the forecast is not specified. Note that for large weather files, the slowdown on the "Location and Resource" page exists on the released version as well.

### CODE REVIEW

File: deploy/runtime/defaults/PV Battery_All Equity Partnership Flip.txt [hunk 1]
```diff
@@ -889,6 +883,12 @@ batt_dispatch_pvs_timestep_multiplier
 1
 1
 10
+batt_dispatch_pvs_wf_forecast_choice
+1
+1
+1
+1
+1.04992e-153
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_All Equity Partnership Flip.txt:891]: #642

File: deploy/runtime/defaults/PV Battery_All Equity Partnership Flip.txt [hunk 2]
```diff
@@ -901,6 +901,12 @@ batt_dispatch_update_frequency_hours
 1
 1
 1
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_All Equity Partnership Flip.txt:909]: #642

File: deploy/runtime/defaults/PV Battery_Commercial.txt [hunk 3]
```diff
@@ -762,6 +762,12 @@ batt_dispatch_excl
 1
 1
 0
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Commercial.txt:770]: #642

File: deploy/runtime/defaults/PV Battery_Host Developer.txt [hunk 4]
```diff
@@ -762,6 +762,12 @@ batt_dispatch_excl
 1
 1
 0
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Host Developer.txt:770]: #642

File: deploy/runtime/defaults/PV Battery_Leveraged Partnership Flip.txt [hunk 5]
```diff
@@ -889,6 +883,12 @@ batt_dispatch_pvs_timestep_multiplier
 1
 1
 10
+batt_dispatch_pvs_wf_forecast_choice
+1
+1
+1
+1
+1.04992e-153
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Leveraged Partnership Flip.txt:891]: #642

File: deploy/runtime/defaults/PV Battery_Leveraged Partnership Flip.txt [hunk 6]
```diff
@@ -901,6 +901,12 @@ batt_dispatch_update_frequency_hours
 1
 1
 1
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Leveraged Partnership Flip.txt:909]: #642

File: deploy/runtime/defaults/PV Battery_Merchant Plant.txt [hunk 7]
```diff
@@ -889,6 +883,12 @@ batt_dispatch_pvs_timestep_multiplier
 1
 1
 10
+batt_dispatch_pvs_wf_forecast_choice
+1
+1
+1
+1
+1.04992e-153
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Merchant Plant.txt:891]: #642

File: deploy/runtime/defaults/PV Battery_Merchant Plant.txt [hunk 8]
```diff
@@ -901,6 +901,12 @@ batt_dispatch_update_frequency_hours
 1
 1
 1
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Merchant Plant.txt:909]: #642

File: deploy/runtime/defaults/PV Battery_Residential.txt [hunk 9]
```diff
@@ -831,6 +831,12 @@ batt_dispatch_excl
 1
 1
 0
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Residential.txt:839]: #642

File: deploy/runtime/defaults/PV Battery_Sale Leaseback.txt [hunk 10]
```diff
@@ -889,6 +883,12 @@ batt_dispatch_pvs_timestep_multiplier
 1
 1
 10
+batt_dispatch_pvs_wf_forecast_choice
+1
+1
+1
+1
+1.04992e-153
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Sale Leaseback.txt:891]: #642

File: deploy/runtime/defaults/PV Battery_Sale Leaseback.txt [hunk 11]
```diff
@@ -901,6 +901,12 @@ batt_dispatch_update_frequency_hours
 1
 1
 1
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Sale Leaseback.txt:909]: #642

File: deploy/runtime/defaults/PV Battery_Single Owner.txt [hunk 12]
```diff
@@ -889,6 +883,12 @@ batt_dispatch_pvs_timestep_multiplier
 1
 1
 3
+batt_dispatch_pvs_wf_forecast_choice
+1
+1
+1
+1
+1.04992e-153
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Single Owner.txt:891]: #642

File: deploy/runtime/defaults/PV Battery_Single Owner.txt [hunk 13]
```diff
@@ -901,6 +901,12 @@ batt_dispatch_update_frequency_hours
 1
 1
 1
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Single Owner.txt:909]: #642

File: deploy/runtime/defaults/PV Battery_Third Party.txt [hunk 14]
```diff
@@ -821,6 +821,12 @@ batt_dispatch_excl
 1
 1
 3.33506e-317
+batt_dispatch_wf_forecast_choice
+1
+1
+1
+1
+3.33505e-317
```
Comment (sjanzou) [inline deploy/runtime/defaults/PV Battery_Third Party.txt:829]: #642

**Review by sjanzou — CHANGES_REQUESTED** [review]
Seems like issue #642 is wreaking havoc with the defaults here! Everything else looks okay!

File: deploy/runtime/defaults/PV Battery_Third Party.txt (hunk 14 above)
Comment (mjprilliman) [reply deploy/runtime/defaults/PV Battery_Third Party.txt:829]: @brtietz Maybe try writing 0.0 as the default to get a true 0. I've had some success with that in the past.

**Review by sjanzou — APPROVED** [review]
Yay! Defaults working now!

**Review by dguittet — APPROVED** [review]

**Pull request merged**